oauth2/src/main/java/com/linecorp/armeria/client/auth/oauth2/AbstractOAuth2AuthorizationGrant.java

```diff
@@ -117,7 +90,101 @@ private static boolean shallObtainInsteadOfUpdate(Throwable throwable) {
      * Validates access token and refreshes it if necessary.
      */
     @Override
-    public final CompletionStage<GrantedOAuth2AccessToken> getAccessToken() {
-        return tokenManager.get();
+    public final CompletionStage<GrantedOAuth2AccessToken> getAccessToken(ClientRequestContext ctx) {
+        final CompletableFuture<GrantedOAuth2AccessToken> future = new CompletableFuture<>();
+        doGetAccessToken(ctx, future, 1);
```

Comment:
nit:
```suggestion
        getAccessToken(ctx, future, 1);
```
We can use the same name because the parameters are different.

Reply:
I've removed this method adopting your approach, thanks! 👍